lib/realtime/repo.ex

```diff
@@ -132,13 +134,33 @@ defmodule Realtime.Repo do
     {:ok, to_sql(:update_all, query)}
   end
 
-  defp run_all_query(conn, query) do
+  defp run_all_query(conn, query, opts \\ []) do
     {query, args} = to_sql(:all, query)
-    Postgrex.query(conn, query, args)
+    run_query_with_trap(conn, query, args, opts)
   end
 
   defp run_delete_query(conn, query) do
     {query, args} = to_sql(:delete_all, query)
-    Postgrex.query(conn, query, args)
+    run_query_with_trap(conn, query, args)
+  end
+
+  defp run_query_with_trap(conn, query, args, opts \\ []) do
```

Comment:
Hey @supabase/dashbit , could you check this section?

We're using Postgrex directly which means that in some occasions we have exceptions that are not properly handled as mentioned in the [query/4](https://hexdocs.pm/postgrex/Postgrex.html#query/4) docs:
```
This function may still raise an exception if there is an issue with types (ArgumentError), connection (DBConnection.ConnectionError), ownership (DBConnection.OwnershipError) or other error (RuntimeError).
```

Added to that I've also noticed that some errors will actually trigger an exit of the process. For example when we're unable to checkout connections from the pool and it [exits the process](https://github.com/elixir-ecto/db_connection/blob/8ef1f2ea54922873590b8939f2dad6b031c5b49c/lib/db_connection/holder.ex#L97) . To cope with all of this situations this was the approach I found since I'm trying to avoid linking processes and "just execute it".

Do you think this is a sane approach? This was based on the docs of [Task](https://hexdocs.pm/elixir/1.14/Task.html )

```
Using Task.Supervisor.async_nolink/2 + Task.yield/2 + Task.shutdown/2 allows you to execute tasks concurrently and retrieve their results or the reason they failed within a given time frame. If the task fails, the caller won't fail. You will receive the error reason either on yield or shutdown.
```

Reply:
Hi @filipecabaco! You don't need the task + trap in this case.

The reason why DBConnection calls exit is to mirror GenServer semantics. In a GenServer, if you try to talk to call a process and it times out, it exists the current process. However, you can catch exits in Elixir:

```
try do
  Postgrex.query(conn, query, args, opts)
catch
  :exit, reason -> ...
rescue
  exception -> ...
end
```

You only need to trap when a linked processes terminates, but that's not the case here. So I think a "defensive" try/catch would do the job here. WDYT?